### ISSUE TITLE
Reason about a configured experiment

### ISSUE THREAD

**Issue by markusdregi** [equinor/ert#2170]
**Introduction**
Currently we pass around a collection of configuration objects in ERT3. [Example:](https://github.com/equinor/ert/blob/0f1b421bd660d902899f9039933882fbe5f09708/ert3/engine/_run.py#L272)

```python
def run(
    ensemble: ert3.config.EnsembleConfig,
    stages_config: ert3.config.StagesConfig,
    experiment_config: ert3.config.ExperimentConfig,
    parameters_config: ert3.config.ParametersConfig,
    workspace_root: pathlib.Path,
    experiment_name: str,
) -> None:
```

I think it makes sense to hold the configuration of an experiment in an object that can represent the entire configuration instead of passing around a collection of individually incomplete objects.

**Comment by markusdregi** [comment]
I'm not arguing that we should carry around a god configuration object. The milestone I'm currently failing to create due to ZenHub issues states that we should reevaluate how far into the code the configuration object is passed. However, I do believe that it makes sense that the workspace (see #2168) serves a god configuration object which by the requester is then translated into an experiment object instead of passed around... 🤷🏻

**Comment by markusdregi** [comment]
Good point @jondequinor! I agree that this approach would have many benefits for which no other approach can provide in a reasonable manner...

I still think gathering the current complexity in ERT3 configuration would be an improvement over the current situation and a better starting point for creating a decoupled user configuration and configuration object in general though🤔

**Comment by jondequinor** [comment]
> My main concern is that we are bypassing the partially untapped resource that is many of our users, and not taking input from them. 

How do you suggest such input is gathered?
